pandas/core/frame.py

```diff
@@ -5349,7 +5349,7 @@ def reindex(
     @overload
     def drop(
         self,
-        labels: IndexLabel = ...,
+        labels: IndexLabel | Index = ...,
```

Comment:
`Index` is not `Hashable` so it isn't part of `IndexLabel`. It might be possible to include `Index` in `IndexLabel` (for now that would create many type errors).

Reply:
Shouldn't this be `| ListLike`? Looks like in the docs/tests any list-like can be passed here